mbus/nats_handler.go

```diff
@@ -142,6 +155,12 @@ func (h *natsHandler) Start(handlerFunc boshhandler.Func) error {
 		nats.ReconnectWait(h.connectRetryInterval),
 		nats.ReconnectJitter(h.maxConnectRetryInterval, h.maxConnectRetryInterval),
 		nats.Secure(connectionInfo.TLSConfig),
+		nats.DisconnectHandler(func(c *nats.Conn) {
+			for c.IsReconnecting() {
+				natsErrorHandlingChan <- fmt.Sprintf("Waiting to reconnect to nats.. Connected: %v", c.IsConnected())
```

Comment:
Could we just inline the `asyncErrorHandler` code here and eliminate all the channel / goroutine stuff?